modules/dnn/src/layers/prior_box_layer.cpp

```diff
@@ -245,6 +246,21 @@ class PriorBoxLayerImpl : public PriorBoxLayer
             _offsetsX.assign(1, offset);
             _offsetsY.assign(1, offset);
         }
+
+#ifdef HAVE_OPENCL
+        Mat offsetsX(1, _offsetsX.size(), CV_32FC1, &_offsetsX[0]);
+        Mat offsetsY(1, _offsetsX.size(), CV_32FC1, &_offsetsY[0]);
+        Mat aspectRatios(1, _aspectRatios.size(), CV_32FC1, &_aspectRatios[0]);
+        Mat variance(1, _variance.size(), CV_32FC1, &_variance[0]);
+
+        offsetsX.copyTo(umat_offsetsX);
+        offsetsY.copyTo(umat_offsetsY);
+        aspectRatios.copyTo(umat_aspectRatios);
+        variance.copyTo(umat_variance);
```

Comment:
BTW, I believe we should not increase memory usage for non-OpenCL backends.
This should be moved into OpenCL specific code (for example, in `forward_ocl()` and called under some "initialization" flag once per layer instance).

Reply:
agree. Such extensive consumption of OpenCL-only memory should not happen by default

Reply:
Fixed in the new update, moved the code into forward_ocl().